src/nix.rs

```diff
@@ -142,7 +143,22 @@ impl ToNickel for rnix::SyntaxNode {
                 "true" => Term::Bool(true),
                 "false" => Term::Bool(false),
                 "null" => Term::Null,
-                id => Term::Var(id.into()),
+                id => {
+                    if state.env.contains(id) || state.with.is_empty() {
+                        Term::Var(id.into())
+                    } else {
+                        Term::App(
+                            mk_app!(
+                                make::op1(
+                                    UnaryOp::StaticAccess("with".into()),
+                                    Term::Var("compat".into()),
+                                ),
```

Comment:
Suggestion: do as for builtin contracts. Create a module named `compat` in the `stdlib` module (the Rust one) that exports this term as `with()`, and just replace this code here `stdlib::compat::with()`, or even `stdlib::compat::with(array)` to avoid cluttering the code and be more resilient to refactoring of the Nickel stdlib.